mlflow/utils/server_cli_utils.py

```diff
@@ -42,10 +42,21 @@ def _is_default_backend_store_uri(backend_store_uri: str) -> bool:
         bool True if the default value is set.
 
     """
-    return backend_store_uri == DEFAULT_TRACKING_URI
+    if backend_store_uri is None:
+        return False
+    return backend_store_uri in {DEFAULT_TRACKING_URI, DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH}
 
 
-def artifacts_only_config_validation(artifacts_only: bool, backend_store_uri: str) -> None:
+def artifacts_only_config_validation(
+    artifacts_only: bool,
+    backend_store_uri: str,
+    enable_workspaces: bool = False,
+) -> None:
+    if artifacts_only and enable_workspaces:
+        # Workspace mode relies on a workspace provider to resolve the default workspace and seed
+        # request context. Artifact-only servers never load that stack, so they cannot determine
+        # the active workspace safely. This decision can be revisited in the future.
+        raise click.UsageError("--enable-workspaces cannot be combined with --artifacts-only.")
```

Comment:
I think this fine to leave for future work if requested. The very nature of this feature stack seems like running an artifacts-only server seems like it's not common for where users would run in this mode.